garglk/sndsdl.c

```diff
@@ -582,36 +732,75 @@ glui32 glk_schannel_play_ext(schanid_t chan, glui32 snd, glui32 repeats, glui32
         case giblorb_ID_FORM:
         case giblorb_ID_AIFF:
         case giblorb_ID_WAVE:
-            return play_sound(chan);
+            result = play_sound(chan);
             break;
 
         case giblorb_ID_OGG:
-            return play_compressed(chan, "OGG");
+            result = play_compressed(chan, "OGG");
             break;
 
         case giblorb_ID_MP3:
-            return play_compressed(chan, "MP3");
+            result = play_compressed(chan, "MP3");
             break;
 
         case giblorb_ID_MOD:
-            return play_mod(chan, len);
+            result = play_mod(chan, len);
             break;
 
         default:
             gli_strict_warning("schannel_play_ext: unknown resource type.");
     }
 
-    return 0;
+    /* if channel was paused it should be paused again */
+    if (result && paused)
+    {
+        if (paused)
+            fprintf(stderr, "glk_schannel_play_ext: pausing channel again\n");
```

Comment:
Is this message necessary?  If so, presumably it should be done via `gli_strict_warning`.